core/HoistComponent.js

```diff
@@ -5,57 +5,78 @@
  * Copyright © 2019 Extremely Heavy Industries Inc.
  */
 import ReactDom from 'react-dom';
-import {XH} from '@xh/hoist/core';
-import {observer} from '@xh/hoist/mobx';
+import {XH, elemFactory, useLoadSupportLinker, useOwnedModelLinker} from '@xh/hoist/core';
+import {useObserver, observer} from '@xh/hoist/mobx';
 import {isPlainObject} from 'lodash';
 import {applyMixin} from '@xh/hoist/utils/js';
 import classNames from 'classnames';
-
 import {ReactiveSupport, XhIdSupport, ManagedSupport} from './mixins';
 
-import {loadSupportLinker}  from './impl/LoadSupportLinker';
 
 /**
- * Core decorator for Components in Hoist.
+ * Core Hoist utility for defining a React function component and corresponding Hoist elemFactory.
+ *
+ * This function always applies the MobX 'observer' behavior to the new component, enabling MobX
+ * powered reactivity and auto-re-rendering. See the hooks package for additional Hoist-provided
+ * custom hooks that can (and should!) be used within function components to replicate the most
+ * essential / relevant capabilities of the class-based HoistComponent decorator below.
  *
- * All React Components in Hoist applications should typically be decorated with this decorator.
- * Exceptions include highly specific low-level components provided to other APIs which may be
- * negatively impacted by the overhead associated with this decorator.
+ * @param {function} renderFn - function defining a React component.
+ * @returns {Object[]} - Array containing the Component and an elemFactory for the Component.
+ *
+ * @see HoistComponent decorator for a ES6 class-based approach to defining a Component in Hoist.
+ */
+export function hoistComponent(renderFn) {
+    const component = observer(renderFn);
+    return [component, elemFactory(component)];
+}
+
+
+/**
+ * Create a Class Component in Hoist.
  *
  * Adds support for MobX reactivity, model awareness, and other convenience methods below.
+ *
+ * NOTE: This decorator provided the original method for specifying class-based components within
```

Comment:
I would like to understand exactly what you lose (or gain) by using a functional component over a class component. (using this comment for my own note taking and thoughts)

ManagedSupport is lost, and seems that the preferred way to handle any additional cleanup in components is to use the `useOnUnmount` hook (which makes perfect sense and is consistent with React best practices). Could see some benefit to a `useManagedState` or similar hook which similar functionality, though in general I think we want to encourage putting state in the model as much as possible

ReactiveSupport is lost, which makes sense since there doesn't seem to be any hook equivalent to a class component constructor (and also should not be needed in most components except in very special cases)

XhIdSupport is lost, and we could write a hook to provide the same functionality, though I've never used it on the component before, only the model. Have there been use-cases where a component id is needed instead of the model id? Don't see how that would be useful on the component since its own key should come from its parent.

Reply:
As Joni Mitchell wrote in 'Both Sides Now'
>>Well something's lost, but something's gained ...  In living every day

Think you pretty well stated things.  I think just staying with the emerging standard React architecture going forward is the overriding value here which trumps a lot of the pros and cons that we could discuss.

Yes, I agree that the change toward functional components ideally obsoletes much of the lost functionality you mention.   The idea is to just aggressively push as much state as possible in to our mobx models, leaving the components as anonymous rendering functions....

So hopefully not too much lost, and wondrous things still to be gained.....